test/extensions/tracers/opentelemetry/grpc_trace_exporter_test.cc

```diff
@@ -70,6 +71,10 @@ TEST_F(OpenTelemetryGrpcTraceExporterTest, CreateExporterAndExportSpan) {
   span.set_name("test");
   *request.add_resource_spans()->add_scope_spans()->add_spans() = span;
   EXPECT_TRUE(exporter.log(request));
+
+  Http::TestRequestHeaderMapImpl metadata;
+  callbacks_->onCreateInitialMetadata(metadata);
+  EXPECT_THAT(metadata.getUserAgentValue(), HasSubstr("OTel-OTLP-Exporter-Envoy/"));
```

Comment:
LGTM for the PR but feels like we should have a test that has some evidence of including version - can we get it to validate it matches the version without hardcoding but using the version from the version utils in the test?

Reply:
I thought somehow I would need to mock the version, but didn't think I could just use it also in the test 😅. Makes total sense, done in [cc3009a](https://github.com/envoyproxy/envoy/pull/32659/commits/cc3009ae1c34303c64b67e022b0c34a7465f3551)